### ISSUE TITLE
Config file default path should not be dev-specific

### ISSUE THREAD

**Issue by dandanlen** [chainflip-io/chainflip-backend#2463]
### Description

The default path for engine config is currently `config/Default.toml`. 

This is confusing since these are default configuration *values*, and it also functions implicitly as a default *path* for the config file. 

As a default file path, `config/Default.toml` it's only useful in a development environment since it's a relative path in the backend repo. 

We should instead store the default values in code (ie. a literal `Settings` struct). 

The config files should instead be named after *environments*, ie. `Development.toml`, `Testing.toml`, `CI.toml`, `Production.toml` etc. and should only include the overrides (ie. the points where local settings differ from the defaults). 

The default *path* to the configuration file should be `/etc/chainflip` and it should be possible to override this with an environment variable `CHAINFLIP_CONFIG_PATH`. 

In summary:
- [ ] Default.toml to be removed in favour of hard-coded `Settings`.
- [ ] Config file default to be named after environment, and include only non-default settings.
- [ ] Default config path to be changed to `/etc/chainflip/config/EngineSettings.toml`, should be configurable via cmd line or environment variable.

CC @ahasna @tomjohnburton
